src/lib/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -949,6 +949,20 @@ public function loadTypesListData(array $typeIds): array
         return $query->execute()->fetchAll();
     }
 
+    public function loadTypesDataByFieldDefinitionIdentifier(string $identifier): array
+    {
+        $query = $this->getLoadTypeQueryBuilder();
+        $query
+            ->where(
```

Comment:
To prevent rewriting the query condition, in case it comes with `getLoadTypeQueryBuilder()` method.
```suggestion
            ->andWhere(
```